### DISCUSSION TITLE
AVRDUDE 8.0 released

### DISCUSSION THREAD

**Discussion by stefanrueger** [avrdudes/avrdude#1912] [Announcements]
I just released version 8.0 of AVRDUDE, a new major release after 7.0 in 2022.

Since the last release 7.3 AVRDUDE has gained
   - Support of multi-memory .hex/.srec files and memory lists including terminal backup and restore commands for all memories
   - New terminal disasm command for disassembling code
   - Strengthened libavrude library (which will probably settle over the next few releases)
   - GUI implementation of AVRDUDE to demonstrate use of the libavrdude library
   - New testing scripts allowing AVRDUDE to test combinations of programmer-chip pairs for all memories
   - Improved infrastructure for cmake and autotools
   - Support for new parts
      + AVR16DU14, AVR16DU20, AVR16DU28, AVR16DU32
      + AVR32DU14, AVR32DU20, AVR32DU28, AVR32DU32
   - Support for new programmers
      + serprog
      + pickit5_updi
   - Support for new serial adapters
      + ch342, ch344, ch347, ch9103

It's been a tremendous effort and particular thanks go to @dl8dtl, @MCUdude, @ndim, @mcuee, @askn37, @funkeleinhorn, @MX682X and countless others for contributions and advice.

Enjoy :)

**Comment by funkeleinhorn** [comment]
Nixpkgs Update: https://github.com/NixOS/nixpkgs/pull/337100

**Comment by dl8dtl** [comment]
Congratulations to the team, and a big "Dankeschön!" to Stefan!

**Comment by mcuee** [comment]
Indeed this is great!
@stefanrueger and @MCUdude contributed most to the avrdude 8.0 release. Thanks!

**Comment by mcuee** [comment]
Good thing is that the following two github actions work nicely.
https://github.com/avrdudes/avrdude/actions/runs/10541592535 (Deploy)
https://github.com/avrdudes/avrdude/actions/runs/10541592513 (avrdude_packing_release).

@umbynos 
Thanks for the help with avrdude_packing_release github action.
* https://github.com/avrdudes/avrdude/pull/1690

**Reply by mcuee** [reply]
Remaining issues -- mainly missing libserialport support for MSVC binaries and avrdude_packing_release build (Static linking build).
* https://github.com/avrdudes/avrdude/issues/1679
* https://github.com/avrdudes/avrdude/issues/1552

**Reply by mcuee** [reply]
Related issue:
* https://github.com/arduino/avrdude-packing/issues/31
* https://github.com/arduino/crossbuild/issues/13

@umbynos @MCUdude @ndim 

We may want to revisit the above issues now that avrdude 8.0 release has been done.

**Comment by mcuee** [comment]
@stefanrueger @MCUdude and @dl8dtl 

I have added the other binaries generated by the above-mentioned github actions in the release.
https://github.com/avrdudes/avrdude/releases/tag/v8.0

**Comment by mcuee** [comment]
My build of avrdude 8.0 using the crossbuild container.
https://github.com/arduino/avrdude-packing/issues/31

1) with libserialport support
[avrdude-v8.0_cross_mingw32.zip](https://github.com/user-attachments/files/16827117/avrdude-v8.0_cross_mingw32.zip)
[avrdude-v8.0_cross_linux_x64.zip](https://github.com/user-attachments/files/16827118/avrdude-v8.0_cross_linux_x64.zip)
[avrdude-v8.0_cross_linux_x86.zip](https://github.com/user-attachments/files/16827129/avrdude-v8.0_cross_linux_x86.zip)
[avrdude-v8.0_cross_linux_arm64.zip](https://github.com/user-attachments/files/16827151/avrdude-v8.0_cross_linux_arm64.zip)
[avrdude-v8.0_cross_linux_arm32.zip](https://github.com/user-attachments/files/16827159/avrdude-v8.0_cross_linux_arm32.zip)

**Comment by mcuee** [comment]
@MCUdude 
Homebrew formula updated -- no support of libserialport and SWIG/Python though.
https://github.com/Homebrew/homebrew-core/blob/c456f8468c12f3b180610b4c78a8f09fd079867d/Formula/a/avrdude.rb

**Comment by gdkadem** [comment]
Autodetect does not work in version 8. When I point the chip manually, it reads and writes without any problems. However, when we want to use automatic detection, it cannot find the chip.

**Reply by mcuee** [reply]
> Autodetect does not work in version 8. When I point the chip manually, it reads and writes without any problems. However, when we want to use automatic detection, it cannot find the chip.

Please help to provide more details. It is better that you create a new issue and provide the details. Thanks.

**Reply by gdkadem** [reply]
When I type "avrdude.exe -c pickit2 -B 4.00 -p m8" without doing anything,
it returns the result "WARNING: Unable to detect MCU".
But when I manually select the processor and select read/write/verify, the process goes smoothly.
"avrdude.exe -c pickit2 -p m8 -B 4.00 -U flash:v:"C:\Users\acer.acer-Computer\Downloads\usbasp.atmega8.2018-06-25.hex":i Verifying 4646 bytes of flash against input file usbasp.atmega8.2018-06-25.hex Reading | ###################################### | 100% 2.12s 4646 bytes of flash verified"

https://github.com/user-attachments/assets/4f33eef1-4cc5-43fe-8ce9-94e7c0a6bdb3

I tried the same thing with USBASP. Same problems occur again.

**Reply by mcuee** [reply]
Please use the command line and not the GUI avrdudess. I think this is a known problem of arvduess.
* https://github.com/ZakKemble/AVRDUDESS/issues/26

**Comment by MCUdude** [comment]
@mcuee I'm thinking about bunding the statically built Avrdude binaries to a new version of my Arduino cores. However, I can't execute the macOS binary, because it has been released by an unauthorized developer. What to do? And how did @umbynos get around this when building "Arduino flavored" Avrdude binaries?

**Reply by mcuee** [reply]
> @mcuee I'm thinking about bunding the statically built Avrdude binaries to a new version of my Arduino cores. However, I can't execute the macOS binary, because it has been released by an unauthorized developer. What to do? And how did @umbynos get around this when building "Arduino flavored" Avrdude binaries?

@MCUdude 

You have to sign up with Apple Developer program and then you can sign the binaries with the Code-Signing Certificate.

Reference:
https://github.com/arduino/avrdude-packing/blob/main/.github/workflows/release.yml

The annual fee is US$99.
https://developer.apple.com/support/enrollment/

Maybe it is not worth the trouble and we may still ask @umbynos for help to release newer version of avrdude binary in the Arduini avrdude-packing git repo.

**Reply by mcuee** [reply]
@MCUdude and @umbynos 

I have added to the comment here about the issue with Code-Signing Certificate for macOS.
* https://github.com/arduino/avrdude-packing/issues/31

**Reply by umbynos** [reply]
Hey, yeah, is because of code-signing. We got a certificate from Apple Dev program and we use that to sign binaries (CLI/IDE)..

**Comment by MCUdude** [comment]
@mcuee libserialport 0.1.2 has just been released, and is available though macOS [Brew](https://formulae.brew.sh/formula/libserialport). Currently, Avrdude is built without libserialport support when using brew. What would it take to fix this?

https://github.com/sigrokproject/libserialport/releases

**Reply by mcuee** [reply]
@MCUdude 

The default github build has libserialport support.

But the static link Arduino avrdude-packing cross build does not have libserialport support. 
* https://github.com/arduino/crossbuild/issues/13

I mentioned in the early comment of this discussion.
* https://github.com/avrdudes/avrdude/discussions/1912#discussioncomment-10509480
> Remaining issues -- mainly missing libserialport support for MSVC binaries and avrdude_packing_release build (Static linking build)

**Reply by mcuee** [reply]
Technically we can build libserialport under the crossbuild container but I have tried it and it does not work.

This was part of the discussion here.
* https://github.com/arduino/avrdude-packing/issues/31

**Reply by MCUdude** [reply]
But a lot of users just use their package manager to install Avrdude. How can we ensure it is built with all necessary dependencies, with libserialport being the latest one?

**Reply by mcuee** [reply]
@MCUdude 

For macOS users, the main package manager is probably Homebrew. There is no issue to ask Homebrew maintinaer to upgrade the formula to include libserialport dependency. You may need to raise a ticket.
https://github.com/Homebrew/homebrew-core/blob/2c65c99efc0565df5972e11d07b175ad7a9d4e18/Formula/a/avrdude.rb 

The other thing is to ask Homebrew maintainer to update libserialport version (maybe they will upgrade soon). You can raise a ticket as well to speed up the adoption.
https://github.com/Homebrew/homebrew-core/blob/2c65c99efc0565df5972e11d07b175ad7a9d4e18/Formula/lib/libserialport.rb

**Reply by mcuee** [reply]
@MCUdude 
You can also raise a PR towards Homebrew project to update the formula.
https://github.com/Homebrew/homebrew-core/pulls

**Reply by mcuee** [reply]
I am not using macPorts but it does include libserialport dependency. But the version is only at avrdude 7.3. And the page mentioned that there is no maintainer.
https://ports.macports.org/port/avrdude/details/

**Comment by mcuee** [comment]
@MCUdude 
Just an update to the macOS libusbserialport on homebrew, now it is using 0.1.2 release.
The commit was on 21-March-2025.
https://github.com/Homebrew/homebrew-core/blob/d090ee24363c91d80af21bdb9a8b5e960a0e7a08/Formula/lib/libserialport.rb
```
class Libserialport < Formula
  desc "Cross-platform serial port C library"
  homepage "https://sigrok.org/wiki/Libserialport"
  url "https://sigrok.org/download/source/libserialport/libserialport-0.1.2.tar.gz"
  sha256 "5deb92b5ca72c0347b07b786848350deca2dcfd975ce613b8e0e1d947a4b4ca9"
  license "LGPL-3.0-or-later"
```

**Reply by MCUdude** [reply]
So libserialport is now a dependency when running `brew install avrdude`?

**Reply by mcuee** [reply]
@MCUdude 

Unfortunately not yet. Maybe you can raise an issue to Homebrew to add libserialport as a dependency to avrdude.

https://github.com/Homebrew/homebrew-core/blob/e4c55d8aa0b95608ad13282eb66ff4fa7b841c83/Formula/a/avrdude.rb
```
  depends_on "cmake" => :build
  depends_on "hidapi"
  depends_on "libftdi"
  depends_on "libusb"
  depends_on "libusb-compat"

  uses_from_macos "bison" => :build
  uses_from_macos "flex" => :build

  on_macos do
    # https://github.com/avrdudes/avrdude/issues/1653
    depends_on "libelf" => :build
  end

  on_linux do
    depends_on "elfutils"
    depends_on "readline"
  end
```

**Reply by MCUdude** [reply]
> Unfortunately not yet. Maybe you can raise an issue to Homebrew to add libserialport as a dependency to avrdude.

I did try to do this and test it locally, but I never managed to work. The best would be if the Homebrew team, or any of the volunteers could do it. @mcuee would you like to create an issue over at the homebrew repo?

**Reply by mcuee** [reply]
@MCUdude 

What do you mean you never managed to get it work? After `brew install libserialport`, your local build of avrdude should have libserialport support. If not, please post your build log. I have no issues to do that.

**Reply by mcuee** [reply]
@chenrui333,

Sorry to disturb you, it seems to me that you are the main contributor to the avrdue Homebrew formula. Just wondering if you can help to add libserialport dependency to the homebrew formula.

Previously libserialport is kind of broken under macOS, witht the newly commited libserialport Homebrew formula, it is good to go. Thanks a lot in advance.

**Comment by mcuee** [comment]
Closing this discussion.
